snippets/csharp/programming-guide/nullable-types/NullableTypesUsage.cs

```diff
@@ -47,22 +51,23 @@ private static void AccessValueOfNullableType()
                 Console.WriteLine("y is undefined");
             }
             // </Snippet3>
-        }
 
-        private static void Conversions()
-        {
             // <Snippet4>
-            int? n = null;
+            int? c = null;
 
-            //int m1 = n;      // Doesn't compile.
-            int n2 = (int)n;   // Compiles, but throws an exception if n is null.
-            int n3 = n.Value;  // Compiles, but throws an exception if n is null.
+            // d = c, if c is not null, d = -1 if c is null.
+            int d = c ?? -1;
+            Console.WriteLine($"d is {d}");
             // </Snippet4>
+        }
 
+        private static void ExplicitCast()
+        {
             // <Snippet5>
-            int? m;
-            int m2 = 10;
-            m = m2;  // Implicit conversion.
+            int? n = null;
+
+            //int m1 = n;    // Doesn't compile.
+            int n2 = (int)n; // Compiles, but throws an exception if n is null.
```

Comment:
Now that pattern matching is a thing, I'd really like to see a sample that does something like this:

```csharp
  public static void Main()
  {
    int? thing = 5;
    int? notAThing = default;

    TestValue(thing);
    TestValue(notAThing);
  }
  private static void TestValue(int? nullableThing)
  {
    if (nullableThing is int number)
    {
      Console.WriteLine(number);
    } else
    {
      Console.WriteLine("thing is default value");
    }

  }
```